projects/dev/src/app/group/group-manage/pending-request/pending-request.component.ts

```diff
@@ -76,11 +149,73 @@ export class PendingRequestComponent implements OnInit, OnChanges {
   onExpandWidth(e) {}
 
   onClickAccept(e) {
-    this.onAccept.emit(e);
+    if (
+      this.selection.length === 0 ||
+      this.acceptLoading ||
+      this.rejectLoading
+    ) {
+      return;
+    }
+
+    this.acceptLoading = true;
+    this.groupService
+      .acceptJoinRequest(
+        this.id,
+        this.selection.map((val) => val.joining_user.group_id)
+      )
+      .subscribe(
+        (res) => {
+          this._manageRequestData(res, "accept", "accepted");
+          this.acceptLoading = false;
+        },
+        (err) => {
+          this._processRequestError(err, "accept");
+          this.acceptLoading = false;
+        }
+      );
   }
 
   onClickReject(e) {
-    this.onReject.emit(e);
+    if (
+      this.selection.length === 0 ||
+      this.acceptLoading ||
+      this.rejectLoading
+    ) {
+      return;
+    }
+
+    this.rejectLoading = true;
+    this.groupService
+      .rejectJoinRequest(
+        this.id,
+        this.selection.map((val) => val.joining_user.group_id)
+      )
+      .subscribe(
+        (res) => {
+          this._manageRequestData(res, "reject", "declined");
+          this.rejectLoading = false;
+        },
+        (err) => {
+          this._processRequestError(err, "reject");
+          this.rejectLoading = false;
+        }
+      );
+  }
+
+  onSelectAll(event) {
+    if (this.selection.length === this.requests.length) {
+      this.selection = [];
+    } else {
+      this.selection = this.requests;
+    }
+  }
+
+  onHeaderCheckboxToggle(event) {
```

Comment:
why is the event called onHeaderCheckboxToggle while it is on footer?

Reply:
That's because in fact it has to be on the top according to the primeng library but I moved it to bottom. (The design)
So the actual event emitter is `onHeaderCheckboxToggle`.
I can change this but need some discussion with you.

Reply:
Ok for the initial even but then you define the binding between the event and the handler in your <app-grid>, this function should have a valid name for the situation. `onSelectAllCheckboxToggle(...)`

Reply:
fixed